server/desktop/src/main/java/dev/slimevr/desktop/platform/SteamVRBridge.kt

```diff
@@ -0,0 +1,426 @@
+import dev.slimevr.tracking.trackers.TrackerRole
+import dev.slimevr.tracking.trackers.TrackerRole.Companion.getById
+import dev.slimevr.tracking.trackers.TrackerUtils.getTrackerForSkeleton
+import dev.slimevr.util.ann.VRServerThread
+import io.eiren.util.collections.FastList
+import solarxr_protocol.rpc.StatusData
+import solarxr_protocol.rpc.StatusDataUnion
+import solarxr_protocol.rpc.StatusSteamVRDisconnectedT
+
+abstract class SteamVRBridge(
+	server: VRServer,
+	threadName: String?,
+	bridgeName: String?,
+	protected val bridgeSettingsKey: String,
+	protected val shareableTrackers: List<Tracker>,
+) : ProtobufBridge(bridgeName),
+	Runnable {
+	protected val runnerThread: Thread = Thread(this, threadName)
+	protected val config: BridgeConfig = server.configManager.vrConfig.getBridge(bridgeSettingsKey)
+
+	@VRServerThread
+	override fun startBridge() {
+		for (tr in shareableTrackers) {
+			val role = tr.trackerPosition?.trackerRole
+			changeShareSettings(
+				role,
+				config.getBridgeTrackerRole(role, false),
+			)
+		}
+		runnerThread.start()
+	}
+
+	@VRServerThread
+	override fun getShareSetting(role: TrackerRole): Boolean {
+		for (tr in shareableTrackers) {
+			if (tr.trackerPosition?.trackerRole == role) {
+				return sharedTrackers.contains(tr)
+			}
+		}
+		return false
+	}
+
+	override fun updateShareSettingsAutomatically(): Boolean {
+		if (!config.automaticSharedTrackersToggling) return false
+		// Enable waist if skeleton has an spine tracker
+		changeShareSettings(TrackerRole.WAIST, instance.humanPoseManager.skeleton.hasSpineTracker)
+
+		// hasChest if waist and/or hip is on, and chest and/or upper chest is also on
+		val hasChest = instance.humanPoseManager.skeleton.let {
+			(it.hipTracker != null || it.waistTracker != null) &&
+				(it.upperChestTracker != null || it.chestTracker != null)
+		}
+		changeShareSettings(TrackerRole.CHEST, hasChest)
+
+		// hasFeet if lower and/or upper leg tracker is on
+		val hasFeet = instance.humanPoseManager.skeleton.let {
+			(it.leftUpperLegTracker != null || it.leftLowerLegTracker != null) &&
+				(it.rightUpperLegTracker != null || it.rightLowerLegTracker != null)
+		}
+		changeShareSettings(TrackerRole.LEFT_FOOT, hasFeet)
+		changeShareSettings(TrackerRole.RIGHT_FOOT, hasFeet)
+
+		// hasKnees if foot tracker and lower and/or upper leg tracker is on
+		val hasKnees = hasFeet &&
+			instance.humanPoseManager.skeleton.let { it.hasLeftFootTracker && it.hasRightFootTracker }
+		changeShareSettings(TrackerRole.LEFT_KNEE, hasKnees)
+		changeShareSettings(TrackerRole.RIGHT_KNEE, hasKnees)
+
+		// hasElbows if an upper arm or a lower arm tracker is on
+		val hasElbows = instance.humanPoseManager.skeleton.let { it.hasLeftArmTracker && it.hasRightArmTracker }
+		changeShareSettings(TrackerRole.LEFT_ELBOW, hasElbows)
+		changeShareSettings(TrackerRole.RIGHT_ELBOW, hasElbows)
+
+		// Hands aren't touched as they will override the controller's tracking
+		return true
+	}
```

Comment:
Why not just keep `instance.humanPoseManager.skeleton` in a local variable? It's so long and you use it a ton here.

Reply:
yeah we could, we no longer reinitialize skeleton anymore, right?